### PR TITLE
[New Model]: Software Information 1.0.0

### LINKED ISSUE #314: [New Model]: Software Information 1.0.0
## Model Description
To ensure the traceability of electronic parts (such as ECUs), it is important to be able to track the associated software information. 

Advantages of a software aspect model:
- no digital twin for each software is necessary (so that it is not necessary to have a separate part number of each used software in a product).
- it is possible to find out what kind of software is installed at a part.

## Relevant Standards
No relevant standards available

## Example Data
```json
{
  "catenaXId": "urn:uuid:454ffa8e-f88d-4ad1-be45-06981756aeb1",
  "softwareInformation": [
    {
     "name": "ECU Software XY12345",
      "softwareId": "SW12345678",
      "version": "1.2.0",
      "lastModifiedOn": "2023-03-21T08:17:29.187+01:00"
    },
    {
      "name": "Software ABC",
    "softwareId": "SW12345678",
      "version": "3.0.0",
      "lastModifiedOn": "2023-03-21T08:17:29.187+01:00"
    }
  ]
}
```

## MS1 Criteria

- [x] The proposed aspect model does not exist already
- [x] The proposed aspect model does not extend an existing aspect but introduces completely new functionality
- [x] The proposal references a Catena-X use case
- [x] Relevant standards are mentioned/linked
- [x] A first draft of the data structure is provided in an example
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by johannsvarela**
## Description
An ECU and installed software components (versions) form a unit consisting of a hardware component (e.g. ECU) and software components for installation in a product (e.g. vehicle). While an ECU as a hardware component for example has a part number and a serial number, there are not always part numbers and serial numbers for each software component.

This leads to the fact that the creation of digital twins for software components is not useful. A mapping of a unit with an ECU part and its installed software components by means of a linkage of digital twins is therefore not possible.

For this reason, the **software must be mapped and documented in a (new) separate aspect model** so that it can be linked to the corresponding digital twin of the ECU.

Closes #314 (old/renamed branch #317)

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.1)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by johannsvarela** [comment]
@bs-jokri @agg3fe would it be possible to review this PR to get the MS2 approval?

File: io.catenax.software_information/1.0.0/SoftwareInformation.ttl
Comment (agg3fe) [inline io.catenax.software_information/1.0.0/SoftwareInformation.ttl:33]: typo in `sofware_information`

File: io.catenax.software_information/1.0.0/SoftwareInformation.ttl
Comment (agg3fe) [inline io.catenax.software_information/1.0.0/SoftwareInformation.ttl:108]: please also add some description here

File: io.catenax.software_information/1.0.0/SoftwareInformation.ttl
Comment (agg3fe) [inline io.catenax.software_information/1.0.0/SoftwareInformation.ttl:50]: typo in 'sofware'.
please check in all other please also if you find any.

File: io.catenax.software_information/1.0.0/SoftwareInformation.ttl
Comment (agg3fe) [inline io.catenax.software_information/1.0.0/SoftwareInformation.ttl:53]: please provide any preferred name here

File: io.catenax.software_information/1.0.0/SoftwareInformation.ttl
Comment (agg3fe) [inline io.catenax.software_information/1.0.0/SoftwareInformation.ttl:63]: there is a shared aspect model available for uuidv4. please check on using that, instead of declaring yourself in model.

**Review by agg3fe — CHANGES_REQUESTED** [review]
please check my comments

**Comment by johannsvarela** [comment]
> please check my comments

@agg3fe can you review my changes i have made? :)

File: io.catenax.software_information/1.0.0/SoftwareInformation.ttl
Comment (agg3fe) [inline io.catenax.software_information/1.0.0/SoftwareInformation.ttl:119]: i think it should be 'end date' and not 'enddate'

**Comment by johannsvarela** [comment]
> @johannsvarela could you please resolve this small PR comment about typo, and then we should be good for MS2 approval.

@agg3fe done :)

**Comment by agg3fe** [comment]
This model is on Hold for now and waiting for some further clarification.

**Comment by johannsvarela** [comment]
> This model is on Hold for now and waiting for some further clarification.

@agg3fe Finally we can continue with the QG process to check and merge this aspect model! :-)

FYI @BenediktMuellerIML

File: io.catenax.software_information/1.0.0/SoftwareInformation.ttl
Comment (agg3fe) [inline io.catenax.software_information/1.0.0/SoftwareInformation.ttl:34]: Can you please uuid 2.0.0 version.
You might need to rebase your branch with main branch.

File: io.catenax.software_information/1.0.0/SoftwareInformation.ttl
Comment (agg3fe) [inline io.catenax.software_information/1.0.0/SoftwareInformation.ttl:2]: 2024 required now

File: io.catenax.software_information/1.0.0/SoftwareInformation.ttl
Comment (agg3fe) [inline io.catenax.software_information/1.0.0/SoftwareInformation.ttl:38]: typo `implented`

File: io.catenax.software_information/RELEASE_NOTES.md
Comment (agg3fe) [inline io.catenax.software_information/RELEASE_NOTES.md:5]: please change the date to 8 April.

Comment (johannsvarela) [reply io.catenax.software_information/RELEASE_NOTES.md]: @agg3fe Done! :)

**Pull request merged**
